backend/compact-connect/lambdas/cloudfront-csp/index.js

```diff
@@ -1,5 +1,337 @@
-// Placeholder for future UI Content Security Policy settings
+//
+//  index.js
+//  CompactConnect
+//
+//  Created by InspiringApps on 7/22/2024.
+//
+
+// ============================================================================
+//                               CONFIGURATION                                =
+// ============================================================================
+/**
+ * Configuration of supported domains per environment.
+ * @type {object}
+ */
+const environments = {
+    csg: {
+        prod: {
+            webFrontend: `app.compactconnect.org`,
+            dataApi: `api.compactconnect.org`,
+            s3Upload: ``, // @TODO: Waiting for environment configuration.
+        },
+        test: {
+            webFrontend: `app.test.compactconnect.org`,
+            dataApi: `api.test.compactconnect.org`,
+            s3Upload: `test-persistentstack-mockbulkuploadsbucket0e8f27eb-ivslg22f2yfz.s3.amazonaws.com`,
+        },
+    },
+    ia: {
+        prod: {
+            webFrontend: `app.jcc.iaapi.io`,
+            dataApi: `api.jcc.iaapi.io`,
+            s3Upload: ``, // @TODO: Waiting for environment configuration.
+        },
+        test: {
+            webFrontend: `app.test.jcc.iaapi.io`,
+            dataApi: `api.test.jcc.iaapi.io`,
+            s3Upload: `test-persistentstack-mockbulkuploadsbucket0e8f27eb-4h1anohxetmp.s3.amazonaws.com`,
+        },
+        justin: {
+            webFrontend: `app.justin.jcc.iaapi.io`,
+            dataApi: `api.test.jcc.iaapi.io`,
+            s3Upload: `test-persistentstack-mockbulkuploadsbucket0e8f27eb-4h1anohxetmp.s3.amazonaws.com`,
+        },
+    },
+};
+
+// ============================================================================
+//                                   HELPERS                                  =
+// ============================================================================
+/**
+ * Get the request domain from the lambda event record.
+ * @param  {object} eventRecord The cloudfront record from the lambda event.
+ * @return {string}             The bare domain of the request domain.
+ */
+const getRequestDomain = (eventRecord) => {
+    let requestDomain = environments.csg.prod.webFrontend;
```

Comment:
Why is `prod` the default?

Reply:
Just so if the lambda event structure changes without us knowing, the Prod CSP headers will be returned; keeping the Prod site functional while we work on an update.

Reply:
makes sense!